packages/frontend/web-ui/src/game/hooks/useJoinExistingGame.spec.ts

```diff
@@ -0,0 +1,792 @@
+          result: null,
+        };
+
+        getUserMeResult = Symbol() as unknown as ReturnType<typeof getUserMe>;
+
+        const payloadActionFixture: PayloadAction<UserMeSerializedResponse> = {
+          payload: {
+            body: {
+              active: true,
+              id: 'id-fixture',
+              name: 'name-fixture',
+            },
+            statusCode: 200,
+          },
+          type: 'sample-type',
+        };
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(authFixture);
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(userFixture);
+
+        (useJoinGame as jest.Mock<typeof useJoinGame>).mockReturnValueOnce(
+          singleApiCallHookResultFixture,
+        );
+
+        (
+          getUserMe as unknown as jest.Mock<typeof getUserMe>
+        ).mockReturnValueOnce(getUserMeResult);
+
+        navigateMock = jest
+          .fn<ReturnType<typeof useNavigate>>()
+          .mockReturnValue(undefined) as ReturnType<typeof useNavigate> &
+          jest.Mock<ReturnType<typeof useNavigate>>;
+
+        (useNavigate as jest.Mock<typeof useNavigate>).mockReturnValueOnce(
+          navigateMock,
+        );
+
+        dispatchMock = jest
+          .fn<ReturnType<typeof useAppDispatch>>()
+          .mockImplementationOnce(
+            <TReturn, TAction>(): TAction | TReturn =>
+              payloadActionFixture as TReturn,
+          ) as ReturnType<typeof useAppDispatch> &
+          jest.Mock<ReturnType<typeof useAppDispatch>>;
+
+        (
+          useAppDispatch as unknown as jest.Mock<typeof useAppDispatch>
+        ).mockReturnValue(dispatchMock);
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(authFixture);
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(userFixture);
+
+        (useJoinGame as jest.Mock<typeof useJoinGame>).mockReturnValueOnce(
+          singleApiCallHookResultFixture,
+        );
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(authFixture);
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(userFixture);
+
+        (useJoinGame as jest.Mock<typeof useJoinGame>).mockReturnValueOnce(
+          singleApiCallHookResultFixture,
+        );
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(authFixture);
+
+        (
+          useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+        ).mockReturnValueOnce(userFixture);
+
+        (useJoinGame as jest.Mock<typeof useJoinGame>).mockReturnValueOnce(
+          singleApiCallHookResultFixture,
+        );
+
+        await waitFor(() => {
```

Comment:
Why shouldn't we use `act` instead?

Reply:
Finally, I put `act` with `async` word inside to allow that all of promises has been resolved. The warning is not appear.